src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        ShopStorage storage = new ShopStorage();
```

Comment:
Instead of creating a new instance of `ShopStorage`, use the singleton instance by calling `ShopStorage.getInstance()`. This ensures that the same storage is used throughout the application, maintaining consistency in the fruit quantities.